### ISSUE TITLE
Redundant newlines when using lists

### ISSUE THREAD

**Issue by seroperson** [emoriarty/notion_to_md#52]
When exported from notion, list on site looks like:
![image](https://github.com/emoriarty/notion_to_md/assets/3791221/2ee637a6-61d3-4bdb-b053-b4a1acad014c)
It looks like this in code:
![image](https://github.com/emoriarty/notion_to_md/assets/3791221/886deafe-b0f6-493f-89b8-2e83a55188ec)
And like this in notion:
![image](https://github.com/emoriarty/notion_to_md/assets/3791221/2b64552e-66ad-4da6-999f-deba54690048)

But I assume it must look like this in code:
![image](https://github.com/emoriarty/notion_to_md/assets/3791221/18d878b0-aad5-479f-9aef-a29b4e1c2971)
And, as a result, like this on the site:
![image](https://github.com/emoriarty/notion_to_md/assets/3791221/d0f8c6e4-fbb6-4655-a28e-cd9668081c6c)

**Comment by emoriarty** [comment]
Hello @seroperson 👋 !

This behavior is in some ways intended intended. You can check [here](https://github.com/emoriarty/notion_to_md/blob/ddfc57c4b8a34644b7c05bcb7fb74d274b512514/lib/notion_to_md/page.rb#L48) why the new lines.

```ruby
def body
  @body ||= blocks.map(&:to_md).compact.join("\n\n")
end
``` 

Basically, every block is treated the same way by the Page object. The page does not distinguish between block, it just join every block with two new lines characters.

The two new lines chars are required because using one one new line char there would be no paragraphs.

Perhaps instead of letting the Page managing how the block are joined, the responsibility of adding the new lines chars can be moved as per block basis. In fact, that makes quite much sense.

I'll give it a thought and be back when I have some conclusions.

But I let you already know that this is not a major issue. The main problem here if you're using Jekyll, is that instead of having a simple list as follows.

```html
<ul>
  <li>Blabla</li>
</ul>
```

This setup adds a paragraph per list item.

```html
<ul>
  <li>
    <p>Blabla</p>
  </li>
</ul>
```

This is quite easy to fix with a little but of CSS. Just modify the margin of the paragraph.

```css
li > p {
  maring-botton: 0;
}
```

I just did the test in the provided example and now looks OK.
<img width="889" alt="Screen Shot 2023-11-18 at 16 45 15" src="https://github.com/emoriarty/notion_to_md/assets/510375/e2cab04d-1a02-4eba-8672-01bc3c3f5221">

Btw, nice [post](https://seroperson.me/2023/11/10/civilization-5-multiplayer-modding/)  😉 .

**Comment by emoriarty** [comment]
Hello @seroperson, this should be fix on version [2.3](https://github.com/emoriarty/notion_to_md/releases/tag/2.3.0). And also a new perk: numbered lists.
